src/test/java/com/wikia/webdriver/common/dataprovider/mobile/MobileAdsDataProvider.java

```diff
@@ -9,119 +9,127 @@
+            "115974612",
+            "50006703732",
+            Arrays.asList(
+                "\"s0\":\"ent\"",
+                "\"s1\":\"_adtest\"",
+                "\"s2\":\"article\"",
+                "\"dmn\":\"wikiacom\"",
+                "\"hostpre\":\"",
+                "\"wpage\":\"synthetictests/dfpparams\"",
+                "\"ref\":\"direct\"",
+                "\"lang\":\"en\"",
+                "\"esrb\":\"teen\""
+            ),
+            Arrays.asList(
+                "\"pos\":\"MOBILE_TOP_LEADERBOARD\"",
+                "\"src\":\"mobile\""
+            )
+        }
+    };
+  }
 
-    @DataProvider
-    public static Object[][] testSynthetic() {
-        return new Object[][]{
-                {"adtest", "SyntheticTests/MobileLeaderboard", "MOBILE_TOP_LEADERBOARD", 320, 100,
-                        136987812,
-                        "mobile", "src/test/resources/adsResources/mobiletl320x100"}
-        };
-    }
+  @DataProvider
+  public static Object[][] testSynthetic() {
+    return new Object[][]{
+        {"adtest", "SyntheticTests/MobileLeaderboard", "MOBILE_TOP_LEADERBOARD", 320, 100,
+         136987812,
+         "mobile", "src/test/resources/adsResources/mobiletl320x100"}
+    };
+  }
 
-    @DataProvider
-    public static Object[][] testDisableGptAds() {
-        return new Object[][]{
-                {
-                        "adtest", "SyntheticTests/ProvidersChain", "InstantGlobals.wgSitewideDisableGpt=1",
-                        "MOBILE_TOP_LEADERBOARD", "mobile; mobile_remnant", ""
-                },
-        };
-    }
+  @DataProvider
+  public static Object[][] testDisableGptAds() {
+    return new Object[][]{
+        {
+            "adtest", "SyntheticTests/ProvidersChain", "InstantGlobals.wgSitewideDisableGpt=1",
+            "MOBILE_TOP_LEADERBOARD", "mobile; mobile_remnant", ""
+        },
+    };
+  }
 
-    @DataProvider
-    public static Object[][] mercuryConsecutivePageViews() {
-        return new Object[][]{
-                {
-                        "adtest", "SyntheticTests/Slots/MercuryOnConsecutivePageViews1",
-                        "SyntheticTests/Slots/MercuryOnConsecutivePageViews2",
-                        "SyntheticTests/Slots/MercuryOnConsecutivePageViews3",
-                        "wka.ent/_adtest//article",
-                }
-        };
-    }
+  @DataProvider
+  public static Object[][] mercuryConsecutivePageViews() {
+    return new Object[][]{
+        {
+            "adtest", "SyntheticTests/Slots/MercuryOnConsecutivePageViews1",
+            "SyntheticTests/Slots/MercuryOnConsecutivePageViews2",
+            "SyntheticTests/Slots/MercuryOnConsecutivePageViews3",
+            "wka.ent/_adtest//article",
+        }
+    };
+  }
+
+  @DataProvider
+  public static Object[][] testAdsHopPostMessage() {
+    return new Object[][]{
+        {"adtest", "DFP/hop", 1, "mobile"},
+        {"adtest", "DFP/hophop", 2, "mobile_remnant"}
```

Comment:
Consider removing magic numbers and using just source to get the the right div you want.